main.tf

```diff
@@ -1,11 +1,11 @@
 locals {
   skip_zone_creation = length(local.zones) == 0
   zones              = var.name == null ? [] : try(tolist(var.name), [tostring(var.name)], [])
-  delegation_set_id  = var.delegation_set_id != "" ? var.delegation_set_id : try(aws_route53_delegation_set.delegation_set[0].id, null)
+  delegation_set_id  = var.delegation_set_id != null ? var.delegation_set_id : try(aws_route53_delegation_set.delegation_set[0].id, null)
 
   run_in_vpc = length(var.vpc_ids) > 0
 
-  skip_delegation_set_creation = local.skip_zone_creation || local.run_in_vpc ? true : var.skip_delegation_set_creation
+  skip_delegation_set_creation = ! var.create || local.skip_zone_creation || local.run_in_vpc ? true : var.skip_delegation_set_creation
```

Comment:
This redundant as we already have skip delegation set.. why do we need create at all?

Reply:
I ditched create. Resolved.